### PR TITLE
Fix the signature for the MS reward

### PR DESCRIPTION
**Pull request by karlb**
The signature is out of sync with the one used in the contracts. The
contract added:

* A message id
* The MSC address

to the signature. Passing the MSC address through the code resulted in
quite a lot of changes. The real change is in `balance_proof.py`

### CODE REVIEW

File: raiden/app.py [hunk 1]
```diff
@@ -70,18 +70,21 @@ def __init__(
         default_secret_registry: SecretRegistry,
         default_service_registry: typing.Optional[ServiceRegistry],
         default_one_to_n_address: typing.Optional[Address],
+        default_msc_address: Address,
         transport,
         raiden_event_handler,
         message_handler,
         user_deposit: UserDeposit = None,
     ):
+        default_msc_address = default_msc_address
```
Comment (palango) [inline raiden/app.py]: what is this for?

File: raiden/raiden_service.py [hunk 2]
```diff
@@ -227,13 +227,17 @@ def update_services_from_balance_proof(
     raiden: "RaidenService",
     chain_state: "ChainState",
     balance_proof: Union[BalanceProofSignedState, BalanceProofUnsignedState],
+    monitoring_service_contract_address: Address,
```
Comment (palango) [inline raiden/raiden_service.py]: the MSC address is stored in the `raiden` instance, so you could use it from there.

File: raiden/app.py (hunk 1 above)
Comment (karlb) [reply raiden/app.py]: That was `default_msc_address = default_monitoring_service_contracts_address` to make the assigment in the following lines fit inside 100 lines. But then I had the same problem in many cases and decided to switch to the shorter name in all place, thus making this line nonsense. Removed!

File: raiden/raiden_service.py (hunk 2 above)
Comment (karlb) [reply raiden/raiden_service.py]: Good point!

**Pull request merged**